### ISSUE TITLE
Sort Dashboard by Publish date where available

### ISSUE THREAD

**Issue by wuz** [thepracticaldev/dev.to#1837]
**Is your feature request related to a problem? Please describe.**
I frequently save article ideas as drafts to my dashboard to come back and work on them. For the most part this works great, but the posts on the dashboard are sorted by creation date (I assume). That leads to a situation where some posts that are published later than others are listed before them on my dashboard. Here is an example:
 
<img width="841" alt="screenshot 2019-02-20 14 50 19" src="https://user-images.githubusercontent.com/2363236/53120072-f8bdf380-351e-11e9-975e-46e5d3fe59f6.png">

As you can see the dates go: "Jan 19, Jan 18, Jan 22, Jan 15". I look at my dashboard to make sure I am publishing regularly, but it is a bit difficult to tell with this sorting.

**Describe the solution you'd like**
Add the ability to sort the dashboard by post publish date, with drafts using their created at date.

**Describe alternatives you've considered**
I could store drafts elsewhere and have considered that, but I love writing in dev.to and like having ideas saved as drafts.

**Comment by maestromac** [comment]
Thank you for this feature request @wuz . This make a lot of sense. Articles should be sorted by publish date and not creation date.

**Comment by jessleenyc** [comment]
Agreed! This is approved if anyone wants to jump on it.

**Comment by maxcell** [comment]
I'd take it on @jessleenyc! 😄

**Comment by maxcell** [comment]
A question that does come to mind when we want to organize everything by published date, which I think I'm down for.  Do we think that drafts should have highest priority in the display? For instance this would be a way of having them:

![image](https://user-images.githubusercontent.com/8431042/53389728-6a060800-395e-11e9-8a28-8a3eb620756b.png)


 I know that you @wuz initially proposed an ability to sort by this published date, do you think other ways of sorting (most comments, most views, etc.) would be valuable enough to justify a way to choose from a dropdown maybe or should we just keep it simple?

**Comment by jessleenyc** [comment]
A way of sorting will likely be most useful but @benhalpern will probably have some additional thoughts there.

**Comment by maxcell** [comment]
I think given the #2134, I will just come up with a process for it and call it a day and let y'all take a gander with it! 😄

**Comment by omenking** [comment]
### Skipping Github Auth
I couldn't remember how to set up the omniauth configuration variables for github so I'm subverted it with my own new quick in. Here if anyone else is feels their spinning their wheels on configuration
here's a shortcut so long as you don't commit it in your PR.

```
# config/routes.rb
get '/qq' => 'pages#quick_in'

# app/controllers/PagesController.rb
def quick_in
	@user = User.first
	sign_in_and_redirect @user, event: :authentication
end

```

If any **core member** could kindly tell me how to get the GITHUB_TOKEN I can video tutoralize
authentication configuration for future contributors.

### The Select Visuals

I could make a good looking select but opted to use what's already within the app to
be consistent with possible existing styling guidelines.

I am using the traditional clearfix element. Don't know if there is a preference for flex,
clearfix:after or clearfix element. I could not determine if from my glance at the existing code.
I'm old school so I used to clearfix element. My apologies if it offends and can change it accordingly.

<img width="888" alt="Screen Shot 2019-04-08 at 10 55 25 AM" src="https://user-images.githubusercontent.com/7776/55734896-5f736f80-59ee-11e9-8f5e-76d3bc902e1d.png">

### Functional Considerations

I figured that we should assume DESC for the most common attributes. I thought some may the ability to
ASC or DESC but keeping it simple here unless told otherwise.

A sort could draft could be included but I thought about it and as a content creator I would like a filter
rather than sort and will consider opening a new feature request after seeing if a ticket already exists.

I will but Recently Created first since it is the current default
<img width="859" alt="Screen Shot 2019-04-08 at 10 55 30 AM" src="https://user-images.githubusercontent.com/7776/55734925-68644100-59ee-11e9-8bd1-60a1e50ec7bf.png">

**Comment by rhymes** [comment]
@omenking 

> how to get the GITHUB_TOKEN I can video tutoralize authentication configuration for future contributors

You can get it from https://github.com/settings/tokens - I'm currently not using one, the app on local works fine without it but I see that the Docker configuration requires one so you might as well set it up I guess

**Comment by Zhao-Andy** [comment]
@omenking, instructions for setting API keys are here: https://docs.dev.to/get-api-keys-dev-env/

For `GITHUB_TOKEN` specifically, see step 6 under "GitHub".

**Comment by jessleenyc** [comment]
Implemented via #2337
